### PR TITLE
Localize itinerary body

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
Requires/~blocked by~ #355 and #358.

This PR adds localization to the itinerary-body package and it components. English and French are provided for illustration. Some unnecessary spaces/span markups have been removed.

What broke:
- Times are displayed in the defaults for the selected locale. There is a way to coerce a time format regardless of the locale, but it has to be done at the `<IntlProvider>` level (and would need a patch PR).
- `timeOptions` props have been removed (see bullet above).
- Props for custom messages (`languageConfig` etc.) have been removed.
- The instruction "Walk/Bicycle x miles to ...": the mode appears in uppercase in the snapshots because the default message is not drawn from a language (that can be changed if people request it).

Commit https://github.com/opentripplanner/otp-ui/pull/357/commits/e5bf9a10ae4a297c6acfcc03d49ed87e19e56608 has the breaking change notation.

Other remarks:
- I could do a separate PR for the `types` package (cf. branch: `update-types`), however because that package is for development time only, that should not be necessary.

### CODE REVIEW

File: packages/itinerary-body/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  /** whether this place row represents the destination */
+  isDestination: boolean;
+  /** Contains details about leg object that is being displayed */
+  leg: Leg;
+}
+
+/**
+ * Shared props for various components that render leg data.
+ */
+export interface LegSharedProps {
```
Comment (miles-grant-ibigroup) [inline packages/itinerary-body/src/types.ts:45]: Could this extend TimeColumnContentProps? It's not semantically sound but might help reduce line count if that's something we're worried about. There's a few fields that are very repeated

**Review by miles-grant-ibigroup — APPROVED** [review]
What a lift! Thanks for what was definitely a lot of work. 

I'm not too fussed about any of the things you mention in the description as no longer working -- none of these feel like showstoppers and I think the flexibility of react-intl obsoletes many of them anyway. As long as we make the release a breaking change we should be fine!

I've left a few comments with slight potential adjustments and clarification, but I'm also happy to merge as-is. Works great locally in the storybook and embedded in otp-rr!

File: packages/itinerary-body/src/AccessLegBody/access-leg-summary.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          id="otpUi.AccessLegBody.summary"
+          values={{
+            // TODO: Implement metric vs imperial (up until now it's just imperial).
+            distance:
+              leg.distance > 0
+                ? humanizeDistanceString(leg.distance, false, intl)
+                : 0,
+            isCarHail: leg.hailedCar,
+            modeId: leg.mode,
+            // FIXME: localize rental device names.
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/AccessLegBody/access-leg-summary.tsx:50]: It's true, I overlooked that.

File: packages/itinerary-body/src/AccessLegBody/rented-vehicle-subheader.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      <FormattedMessage
+        defaultMessage={
+          defaultMessages[
+            "otpUi.AccessLegBody.RentedVehicleSubheader.pickupRental"
+          ]
+        }
+        description="Instructs to pick up a rental vehicle"
+        id="otpUi.AccessLegBody.RentedVehicleSubheader.pickupRental"
+        values={{
+          company,
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/AccessLegBody/rented-vehicle-subheader.tsx:80]: I think it is possible, although I am not sure we want to do that.

File: packages/itinerary-body/src/AccessLegBody/leg-diagram-preview.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+              defaultMessage={
+                defaultMessages[
+                  "otpUi.AccessLegBody.LegDiagramPreview.elevationChart"
+                ]
+              }
+              description="Title text for elevation chart"
+              id="otpUi.AccessLegBody.LegDiagramPreview.elevationChart"
+            />
+            <S.PreviewDiagramElevationGain>
+              ↑{this.formatElevation(profile.gain * METERS_TO_FEET)}
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/AccessLegBody/leg-diagram-preview.tsx:119]: We'll have to provide a new prop or so (in a different PR) to switch between units.

File: packages/itinerary-body/src/AccessLegBody/access-leg-summary.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/AccessLegBody/access-leg-summary.tsx]: Updated in https://github.com/opentripplanner/otp-ui/pull/357/commits/eaddcca230afea2c21130041f81b0049bae8476d.

File: packages/itinerary-body/src/otp-react-redux/view-stop-button.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  onClick = (): void => {
+    const { onStopClick, stopId } = this.props;
+    onStopClick({ stopId });
+  };
+
+  render(): ReactElement {
+    const { text } = this.props;
+    return (
+      <S.ViewerButton onClick={this.onClick}>
+        {text || (
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/otp-react-redux/view-stop-button.tsx]: Good find! I removed that prop in https://github.com/opentripplanner/otp-ui/pull/357/commits/ce9e970732dc074628cf8d9093343a8a74758b7a.

File: packages/itinerary-body/src/types.ts (hunk 1 above)
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/types.ts]: See if https://github.com/opentripplanner/otp-ui/pull/357/commits/34ff0279e9ba5e10d012a29d7b79933f15c71abb makes things better or worse...

File: packages/itinerary-body/src/AccessLegBody/leg-diagram-preview.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/AccessLegBody/leg-diagram-preview.tsx:119]: I was thinking you could do it within the language configuration but I think you're right a new prop makes more sense, but not in this PR!

File: packages/itinerary-body/src/AccessLegBody/rented-vehicle-subheader.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/AccessLegBody/rented-vehicle-subheader.tsx:80]: Sounds good!

File: packages/itinerary-body/src/types.ts (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/types.ts]: I think I prefer it, but I'll leave this unresolved for the second reviewer to take a look at it.

**Comment by binh-dam-ibigroup** [comment]
This is ready for review with changes needed for #366, but right now tests are failing because of an error in the `from-to-picker` package, see https://github.com/opentripplanner/otp-ui/runs/5771498011?check_suite_focus=true#step:8:130, so this might be blocked until that package is fixed.

**Comment by miles-grant-ibigroup** [comment]
This looks to me like it might've been a bad merge. At least some changes in those snapshots are changes to IDs I made during the stopId/stopCode PR.

**Comment by binh-dam-ibigroup** [comment]
> This looks to me like it might've been a bad merge. At least some changes in those snapshots are changes to IDs I made during the stopId/stopCode PR.

I guess you are right. Scrap what I said before, snapshot is fixed now and PR is ready for review.

**Review by demory — APPROVED** [review]

**Pull request merged**